collector.spec.in

```diff
@@ -33,22 +37,68 @@ mkdir -p _build/bin
 %setup -q -n %{name}-%{version}
 
 %build
+
 go build -v -buildmode pie -mod vendor -o %{gobuilddir}/bin/opentelemetry-collector
+mv opentelemetry-collector-with-options %{gobuilddir}/bin/opentelemetry-collector-with-options
```

Comment:
I ran the build through fedpkg and I had no issues replacing this line and adding it to line 61.
```suggestion
install -m 0755 -p opentelemetry-collector-with-options %{buildroot}%{_bindir}/
```
I remember that you had issues using install to copy this file to bin but there might have been a different root cause for this.

Reply:
But the final installation happens then in the install section, right?
![image](https://github.com/os-observability/redhat-opentelemetry-collector/assets/10403402/8d2b3f20-605b-4057-9a2a-3f29df2ddad0)
Means we would call install twice. Isnt that a bit misleading?

Reply:
yes that is correct so basically this:
```
# install files
install -p -m 0644  ./00-default-receivers.yaml %{buildroot}%{_sysconfdir}/opentelemetry-collector/configs/00-default-receivers.yaml
install -p -m 0644  ./opentelemetry-collector.service %{buildroot}%{_unitdir}/%{name}.service
install -p -m 0644  ./microshift-opentelemetry-gateway.yaml %{buildroot}%{_sysconfdir}/microshift/manifests/opentelemetry-collector/manifest.yaml
install -p -m 0644  ./kustomization.yaml %{buildroot}%{_sysconfdir}/microshift/manifests/opentelemetry-collector/kustomization.yaml

install -m 0755 -vd                     %{buildroot}%{_bindir}
install -m 0755 -vp %{gobuilddir}/bin/* %{buildroot}%{_bindir}/
install -m 0755 -p opentelemetry-collector-with-options %{buildroot}%{_bindir}/
```
The line above shouldn't be put into line 42.